_includes/devdoc/ref_p2p_networking.md

```diff
@@ -167,6 +168,88 @@ different reasons:
+the corresponding transaction either from the blocktxn message or from other
+sources and place it in the first available position in the block.
+
+{% endautocrossref %}
+
+#### CmpctBlock
+{% include helpers/subhead-links.md %}
+
+{% autocrossref %}
+
+*Added in protocol version 70014 as described by BIP152.*
+
+A `cmpctblock` message is a compact enocoding of a new block, used to relay
+blocks with faster propogation and lower bandwidth than relaying the full
+block. See `sendcmpct` for details of compact block relay modes.
+
+`cmpctblock` contains the block header, short transaction ids for the
+transactions which the node expects the peer to already have, and
+PrefilledTransaction structures for the other transactions in the block
+(which the node expects the peer to not yet have).
+
+| Bytes                   | Name                 | Data Type                     | Description
+|-------------------------|----------------------|-------------------------------|----------------
+| 80                      | header               | Block header                  | The header of the block being provided. The first 80 bytes of the block as defined by the encoding used by "block" messages.
+| 8                       | nonce                | uint64_t                      | A nonce for use in short transaction ID calculations.
+| *Varies* (1 or 3 bytes) | shortids_length      | compactSize uint              | The number of short transaction IDs (ie block tx count - prefilledtxn_length).
+| *Varies*                | shortids             | List of 6-byte integers       | The short transaction IDs calculated from the transactions which are not provided explicitly in prefilledtxn. The construction of the short transaction ID is described below.
+| *Varies* (1 or 3 bytes) | prefilledtxn_length  | compactSize uint              | The number of prefilled transactions in prefilledtxn (ie block tx count - shortids_length).
+| *Varies*                | prefilledtxn         | List of PrefilledTransactions | The transactions which the node expects the peer to be missing. Always includes the coinbase transaction and may include additional transactions in the block. The format of the PrefilledTransaction structure is defined below.
+
+The Prefilled Transaction structure is:
+
+| Bytes             | Name        | Data Type        | Description
+|-------------------|-------------|------------------|----------------
+| *Varies* (1 or 3) | index       | compactSize uint | The index of the transaction in the block, differentially encoded.
+| *Varies*          | Transaction | Transaction      | The transaction being relayed in raw transaction form.
+
+The indexes are *differentially encoded*. Instead of using raw indexes, the
+number encoded is the difference between the current index and the previous
+index, minus one. For example, a first index of 0 implies a real index of 0,
+a second index of 0 thereafter refers to a real index of 1, etc.
```

Comment:
index 0 is the coinbase transaction (which will always be included in a cmpct block message, since the peer would never have the coinbase transaction in their mempool). I've added a note to make that clear.